packages/edit-site/src/components/navigation-sidebar/navigation-toggle/index.js

```diff
@@ -83,4 +83,4 @@ function NavigationToggle( { icon, isOpen } ) {
 	);
 }
 
-export default NavigationToggle;
+export default withFilters( 'editSite.NavigationToggle' )( NavigationToggle );
```

Comment:
For me it's not clear why we need this filter at all. and we've been moving away from withFilters and addFilter in Gutenberg. I know folks in WordPress want to be able to tweak everything in the UI but I don't think that's a healthy for the project in the long term.

Reply:
I don't think Slot and Fill is any better here personally, it's not clear what the slot would be, no semantics or anything. To be honest, I don't know what to do here.

Reply:
>  I know folks in WordPress want to be able to tweak everything in the UI but I don't think that's a healthy for the project in the long term.

I agree, it doesn't seem healthy.  Adding tools to bypass UI because it isn't in the state we would like it to be, instead of taking the time to update the desired state seems unhealthy.  Also both filter and slot/fill feel a bit odd on this toggle component.

Would a slot/fill make more sense on the panel content?  If we want to abstract the template hierarchy away from new users, it would be beneficial to be able to remove the templates section from the navigation panel.  I think it was with this idea that @copons created a draft PR for adding a Slot/Fill for the actual sidebar content (https://github.com/WordPress/gutenberg/pull/33298).  To me, that makes a bit more sense for a slot/fill than this toggle button, but I wonder your thoughts on that as well?  It would still be good to update the navigation panel in a way that is not overwhelming to new users while advanced features are still accessible for power users and theme developers, but if were running a platform where we are trying to handle the template hierarchy for the user then hiding that section altogether seems necessary.

Reply:
@Addison-Stavlo I think that I've figured out better ways to SlotFill that's more respectful of Core WP.

Basically, instead of SlotFill-ing the entire content, we should just simply add a SlotFill _after_ it, so that folks can add other menus and whatnot.
If folks don't need the vanilla menus, they can probably just hide them via CSS for now.

This about the content would be a temporary stopgap anyway, while we work on improving the menus for everybody.